lib/rails_state_machine/state_machine.rb

```diff
@@ -132,12 +132,12 @@ def define_event_methods
       event_names.each do |event_name, event|
         model_module_eval do
           define_method "#{event_name}" do |**attributes|
-            prepare_state_event_change(attributes.merge("#{state_attribute}_event": event_name))
+            state_machine_state_manager(state_attribute).transition_to(event_name, raise_for_invalid_transitions: true)
             save
           end
 
           define_method "#{event_name}!" do |**attributes|
-            prepare_state_event_change(attributes.merge("#{state_attribute}_event": event_name))
+            state_machine_state_manager(state_attribute).transition_to(event_name, raise_for_invalid_transitions: true)
             save!
           end
```

Comment:
IMO the change being introduced contradicts the previous/initial idea of `event_name` vs `event_name!`.

The bang variant calls `save!` and already raises an error if the record cannot be saved.
Previously, an invalid transition always raised an error but if this is still the default for the non-bang variant, I believe that's likely never expected or useful.

I believe it's sensible to imply not raising an error for `event_name` when an invalid transition is being called.